modules/dxkultureBidAdapter.js

```diff
@@ -138,33 +138,91 @@ export const spec = {
 
   getUserSyncs: function (syncOptions, serverResponses, gdprConsent, uspConsent) {
     logInfo('dxkulture.getUserSyncs', 'syncOptions', syncOptions, 'serverResponses', serverResponses);
-
     let syncs = [];
 
     if (!syncOptions.iframeEnabled && !syncOptions.pixelEnabled) {
       return syncs;
     }
 
-    if (syncOptions.iframeEnabled || syncOptions.pixelEnabled) {
-      let pixelType = syncOptions.iframeEnabled ? 'iframe' : 'image';
-      let queryParamStrings = [];
-      let syncUrl = SYNC_URL;
-      if (gdprConsent) {
-        queryParamStrings.push('gdpr=' + (gdprConsent.gdprApplies ? 1 : 0));
-        queryParamStrings.push('gdpr_consent=' + encodeURIComponent(gdprConsent.consentString || ''));
-      }
-      if (uspConsent) {
-        queryParamStrings.push('us_privacy=' + encodeURIComponent(uspConsent));
+    serverResponses.forEach(resp => {
+      const userSync = deepAccess(resp, 'body.ext.usersync');
+      if (userSync) {
+        let syncDetails = [];
+        Object.keys(userSync).forEach(key => {
+          const value = userSync[key];
+          if (value.syncs && value.syncs.length) {
+            syncDetails = syncDetails.concat(value.syncs);
+          }
+        });
+        syncDetails.forEach(syncDetails => {
+          let queryParamStrings = [];
+          let syncUrl = syncDetails.url;
+
+          if (syncDetails.type === 'iframe') {
+            if (gdprConsent) {
+              queryParamStrings.push('gdpr=' + (gdprConsent.gdprApplies ? 1 : 0));
+              queryParamStrings.push('gdpr_consent=' + encodeURIComponent(gdprConsent.consentString || ''));
+            }
+            if (uspConsent) {
+              queryParamStrings.push('us_privacy=' + encodeURIComponent(uspConsent));
+            }
+            syncUrl = `${syncDetails.url}${queryParamStrings.length > 0 ? '?' + queryParamStrings.join('&') : ''}`
+          }
+
+          syncs.push({
+            type: syncDetails.type === 'iframe' ? 'iframe' : 'image',
+            url: syncUrl
+          });
+        });
+
+        if (!syncOptions.iframeEnabled) {
+          syncs = syncs.filter(s => s.type !== 'iframe')
+        }
+        if (!syncOptions.pixelEnabled) {
+          syncs = syncs.filter(s => s.type !== 'image')
```

Comment:
I just want to make sure this was intended to overwrite as pixel will take priority over iframe. Either way, can this be an if else as to not run both conditions

Reply:
@mmoschovas Thank you for your suggestion. We've addressed the issue by updating the code to match the logic.